patches/android_webview-browser-aw_permission_manager.cc.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/android_webview/browser/aw_permission_manager.cc b/android_webview/browser/aw_permission_manager.cc
+index f1979fdcc078f31d8caad5da6cd68906b138f263..cc68cf801b05615590f187123bb9aafd30521518 100644
+--- a/android_webview/browser/aw_permission_manager.cc
++++ b/android_webview/browser/aw_permission_manager.cc
+@@ -335,6 +335,7 @@ int AwPermissionManager::RequestPermissions(
+       case PermissionType::BACKGROUND_FETCH:
+       case PermissionType::IDLE_DETECTION:
+       case PermissionType::PERIODIC_BACKGROUND_SYNC:
++      case PermissionType::AUTOPLAY:
```

Comment:
patches should almost always use `define` to make them extensible. See `permission_controller_impl.cc`, but is this the right place to put it? We do have permission requests for autoplay afaik cc @darkdh

Reply:
On desktop, we will convert `PermissionType::AUTOPLAY` to `CONTENT_SETTINGS_TYPE_AUTOPLAY` (`PermissionTypeToContentSetting`) and then have a general way to handle it through content setting. On Android, it seems like we have to specifically define and implement virtual function in `AwBrowserPermissionRequestDelegate` for each support type.

Reply:
@bridiver @darkdh 
Updated the patch to use defines.

As for special processing `PermissionType::AUTOPLAY`, this `AwPermissionManager::RequestPermissions` eventually goes to WebViewContentsClientAdapter.onPermissionRequest (https://cs.chromium.org/chromium/src/android_webview/glue/java/src/com/android/webview/chromium/WebViewContentsClientAdapter.java?l=562) , and then if WebView consumer had set up the `WebChromeClient`,  ```mWebChromeClient.onPermissionRequest(adapter)``` will be invoked. Owner of `WebChromeClient` can then call `permissionRequest.grant` or `permissionRequest.deny`.

Currently Chromium WebView files aw_*.* are included into the build, but it is not available to set as a system webview, also it is hard to believe, that android app developers who use WebView would expect the Brave WebView  would be a system WebView and apps have some special handling for `AUTOPLAY` permission.

So `AUTOPLAY` is permissions in this patch remains denied.